docs/fides/docs/development/code_style.md

```diff
@@ -2,35 +2,98 @@
 
 ---
 
-## Black formatting
+## General
 
-Fides's code is formatted using the [black](https://github.com/ambv/black) style. This style is checked in a CI step, and merges to master are prevented if code does not conform.
+### Docstrings
 
-To apply black to your code, run black from the root Fidesctl directory:
+Docstrings are required for every function, class and method. No specific style is required or encouraged, as we expect that most of the relevant information can be gleaned from both the function signature's type-hints as well as descriptive parameter names. The docstring should serve to give additional context/flavour beyond that which can be gained from the code itself.
 
-```bash
-cd fidesctl
-black .
+```python title="Docstring Example"
+# Bad
+def execute_evaluation(taxonomy: Taxonomy) -> Evaluation:
+    """
+    Execute an evaluation. 
+    """
+
+# Good
+def execute_evaluation(taxonomy: Taxonomy) -> Evaluation:
+    """
+    Check the stated constraints of each Privacy Policy's rules against
+    each system's privacy declarations.
+    """
 ```
 
-A number of extensions are available for popular editors that will automatically apply black to your code.
+### Variable/Parameter Naming
+
+Variable and parameter names should be as self-describing as possible. Brevity is not a concern here. Here are some common examples for writing good self-documenting code:
 
-## Pylint
+```python title="Single Letter Variable Names"
+# Incorrect
+for k, v in example_dict.items():
+    print(k)
+    print(v)
 
-Fides's code is linted using [pylint](https://pylint.org/). Linter checks run as part of a CI step and merges to master are prevented if code does not conform.
+# Correct
+for key, value in example_dict.items():
+    print(key)
+    print(value)
+```
+
+```python title="Abbreviated Variable Names"
+# Incorrect
+r = requests.get(url)
 
-To apply pylint to your code, run pylint from the root Fidesctl directory:
+# Incorrect
+resp = reqeusts.get(url)
 
-```bash
-cd fidesctl
-pylint src
+# Correct
+response = requests.get(url)
 ```
 
-## Mypy typing
+```python title="Type Ambiguous Variable Names"
+# Incorrect
+food = ["apple", "banana"] 
+
+# Incorrect
+foods = ["apple", "banana"] 
+
+# More Correct
+# Use type annotations if the name is somewhat ambiguous
+foods: List[str] = ["apple", "banana"] 
+
+# Most Correct
+# The type is contained in the name
+foods_list = ["apple", "banana"] 
```

Comment:
we don't enforce type hints outside of function parameters. maybe we can change both methods to `correct`, or do we want to drive a stake in the ground and say that _all_ variables need typing?

Reply:
When it comes to strong vs. weak typing, IMO, there's no such thing as strong enough. If MyPy supports it, we should include as much type checking and enforcement as we can.

Reply:
Python isn't a strongly-typed language though, and adding additional requirements around typing can slow down development and scare away contributors. I don't know of a single OSS python project that requires typing on _every_ variable. the standard practice is to only require it on function parameters/return values. I vote for not requiring it, and either having it in the name _or_ as an explicit type hint

@seanpreston @pattisdr @ethyca/fides-control thoughts here on mypy requirements?

Reply:
Regarding the type in the variable name, I do lean towards including the type in the variable name but I would be okay with not making it an explicit recommendation. Seeing a variable `foods_list` allows me to quickly identify that I am working with a list, but like Phil mentioned, we do have the type available. There is value in keeping this page brief so that our recommendations are taken seriously, so I would think that we can reserve it for strong/required recommendations.

Reply:
i'll add a few that are just "correct" as examples to show that any are ok as long as they follow one of those styles, since it would need to be either one of those for me to approve it. thanks for the feedback all!

Reply:
late comment, but I agree with you @ThomasLaPiana. I love type hints, we should use them everywhere we can, but I wouldn't enforce them across the board:

https://www.python.org/dev/peps/pep-0563/#non-goals

> Just like in PEP 484 and PEP 526, it should be emphasized that Python will remain a dynamically typed language, and the authors have no desire to ever make type hints mandatory, even by convention.